preset/rabbitmq/preset.go

```diff
@@ -121,6 +137,148 @@ func (p *P) setDefaults() {
 	}
 }
 
+func (p *P) loadFiles() error {
+	if len(p.MessagesFiles) > 0 {
+		for _, fName := range p.MessagesFiles {
+			msgs, err := p.loadMessagesFromFile(fName)
+			if err != nil {
+				return fmt.Errorf("can't read messages from file '%s': %w", fName, err)
+			}
+
+			p.Messages = append(p.Messages, msgs...)
+		}
+	}
+
+	return nil
+}
+
+func declareQueues(ch *amqp.Channel, qs []string) error {
+	for _, queue := range qs {
+		_, err := ch.QueueDeclare(queue, false, false, false, false, nil)
```

Comment:
I wonder what's the difference between creating queues explicitly (like here) and creating them implicitly when publishing messages. I haven't look into it yet, but I'd like to understand: is `declareQueues` a necessary step in rabbitmq state setup?

If it makes a positive difference, let's keep it, but if it's the same as just using "create on publish" feature, let's get rid of some more code.

Reply:
I wrote `declareQueues` because linter started throwing errors that `initf` was too long (more than 60 loc). Regarding the queue declaration itself, it's required if the queue doesn't already exist with those options. If it does, `QueueDeclare` returns the same data it would return when it's declared for the first time (information about the queue). Once it's been declared for the first time, there is no need to declare it again. Publish just publishes the message to queue and consume just consumes.